pkg/apply/apply.go

```diff
@@ -51,14 +52,12 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 	if err := MergeObjectForUpdate(existing, obj); err != nil {
 		return errors.Wrapf(err, "could not merge object %s with existing", objDesc)
 	}
-
-	if err := client.Update(ctx, obj); err != nil {
-		return errors.Wrapf(err, "could not update object %s", objDesc)
-	}
-	if existing.GetResourceVersion() == obj.GetResourceVersion() {
-		log.Print("update was noop")
-	} else {
-		log.Print("update was successful")
+	if !equality.Semantic.DeepEqual(existing, obj) {
```

Comment:
I am not sure this function achive the objective.
can you please add a unit  test for this.

Reply:
Uh, that will be hard. We need mocking for this function, but can try

Reply:
It's not must I can live with that